torchaudio/models/decoder/_ctc_decoder.py

```diff
@@ -28,21 +28,23 @@
         load_words as _load_words,
     )
 except Exception:
-    torchaudio._extension._load_lib("libtorchaudio_decoder")
-    from torchaudio._torchaudio_decoder import (
-        _create_word_dict,
-        _CriterionType,
-        _Dictionary,
-        _KenLM,
-        _LexiconDecoder,
-        _LexiconDecoderOptions,
-        _LexiconFreeDecoder,
-        _LexiconFreeDecoderOptions,
-        _LM,
-        _load_words,
-        _SmearingMode,
-        _Trie,
-        _ZeroLM,
+    torchaudio._extension._load_lib("libflashlight-text")
```

Comment:
would fbcode be able to directly use this version as well or do we need to keep the separate branch above?

Reply:
+1

Reply:
We need to keep the branch.

In fbcode, flashlight-text and torchaudio are logically separated during the whole lifecycle, from source to deployment package. So it is not possible for torchaudio to have decoder code inside of torchaudio package. Instead, flashlight-text is always available, so the decoder code is imported from upstream flashlight packaget.

Reply:
BTW in OSS, the situation is different. I think there are main three audience we need to think;

1. users who only use torchaudio
2. users who use flashlight-text and torchaudio
3. users who develops flashlight-text and torchaudio.

I think the majority of torchaudio users are in category 1. and they do not import the native flashlight-text, they will use the one torchaudio ships. This is straightforward.
The users in category 2, if torchaudio attempts to import the version it ships, then importing both flashlight-text and torchaudio become impossible due to duplicated symbols in both binaries. This is why I put torchaudio's version in lower priority, and attempt to import flashlight first.
Those in category 3, if there is a sudden backward-incompatible change, (say LM attribute from flashlight binding becomes unavailable in favor of some new attribute) then using torchaudio with flashlight become impossible. This can be only prevented by flashlight-text provides soft BC-breaking change and migration period. But Jacob will provide guidance if this ever happens, so it should be fine.